dataflow/src/main/java/org/checkerframework/dataflow/analysis/FlowExpressions.java

```diff
@@ -1057,6 +1074,108 @@ public String toString() {
         }
     }
 
+    /** Receiver takes in binary operations. */
+    public static class BinaryAccess extends Receiver {
+
+        /** The binary node. */
+        protected final BinaryOperationNode node;
+        /** Receiver of the left operator. */
+        protected final Receiver left;
+        /** Receiver of the right operator. */
+        protected final Receiver right;
+
+        /** Constructor, takes in the BinaryOperationNode and the left and right Receiver. */
+        public BinaryAccess(
+                TypeMirror type, BinaryOperationNode node, Receiver left, Receiver right) {
+            super(type);
+            this.node = node;
+            this.left = left;
+            this.right = right;
+        }
+
+        /** Get the binary operation node. */
+        public BinaryOperationNode getNode() {
+            return node;
+        }
+
+        /** Get the left receiver. */
+        public Receiver getLeft() {
+            return left;
+        }
+
+        /** Get the right receiver. */
+        public Receiver getRight() {
+            return right;
+        }
+
+        @Override
+        public boolean containsOfClass(Class<? extends Receiver> clazz) {
+            if (getClass().equals(clazz)) {
+                return true;
+            }
+            return right.containsOfClass(clazz) || left.containsOfClass(clazz);
+        }
+
+        @Override
+        public boolean isUnassignableByOtherCode() {
+            return right.isUnassignableByOtherCode() && left.isUnassignableByOtherCode();
+        }
+
+        @Override
+        public boolean isUnmodifiableByOtherCode() {
+            return right.isUnmodifiableByOtherCode() && left.isUnmodifiableByOtherCode();
+        }
+
+        @Override
+        public boolean syntacticEquals(Receiver other) {
+            if (!(other instanceof BinaryAccess)) {
+                return false;
+            }
+            if (!node.getClass().equals(((BinaryAccess) other).node.getClass())) {
+                return false;
+            }
+            return right.syntacticEquals(((BinaryAccess) other).right)
+                    && left.syntacticEquals(((BinaryAccess) other).left);
+        }
+
+        @Override
+        public boolean containsModifiableAliasOf(Store<?> store, Receiver other) {
+            return right.containsModifiableAliasOf(store, other)
+                    || left.containsModifiableAliasOf(store, other);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(node, left, right);
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (!(other instanceof BinaryAccess)) {
+                return false;
+            }
+            if (!node.getClass().equals(((BinaryAccess) other).node.getClass())) {
+                return false;
+            }
+            if (node instanceof BitwiseAndNode
```

Comment:
Introduce a variable for `(BinaryAccess) other`, to make the expressions below simpler.